modules/gdnative/godot/godot_string.cpp

```diff
@@ -112,6 +112,188 @@ void GDAPI godot_string_destroy(godot_string *p_self) {
 	self->~String();
 }
 
+/* Helpers */
+
+godot_bool GDAPI godot_string_begins_with(const godot_string *self, const godot_string *string) {
+    const String *_self = (const String *) self;
+    const String *_string = (const String *) string;
+
+    return _self->begins_with(*_string);
+}
+
+godot_bool GDAPI godot_string_begins_with_char_array(const godot_string *self, const char *char_array) {
+    const String *_self = (const String *) self;
+
+    return _self->begins_with(char_array);
+}
+
+// /?\ How to convert a 'Vector<String>' to 'const Array&' ?
+/*
+godot_array GDAPI godot_string_bigrams(const godot_string *self) {
+    godot_array bigrams;
+    const String *_self = (const String *) self;
+    memnew_placement(&bigrams, Array(_self->bigrams()));
+
+    return bigrams;
+}
+*/
+
+godot_bool GDAPI godot_string_ends_with(const godot_string *self, const godot_string *string) {
+    const String *_self = (const String *) self;
+    const String *_string = (const String *) string;
+
+    return _self->ends_with(*_string);
+}
+
+godot_string GDAPI godot_string_format(const godot_string *self, const godot_variant *values) {
+    return godot_string_format_with_custom_placeholder(self, values, "{_}");
+}
+
+godot_string GDAPI godot_string_format_with_custom_placeholder(const godot_string *self, const godot_variant *values, const char *placeholder) {
+    const String *_self = (const String *) self;
+    const Variant *_values = (const Variant *) values;
+    String _placeholder = String(placeholder);
+    godot_string result;
+    memnew_placement(&result, String(_self->format(*_values, _placeholder)));
+
+    return result;
+}
+
+godot_string GDAPI godot_string_insert(const godot_string *self, godot_int at_pos, godot_string content) {
+    const String *_self = (const String *) self;
+    String *_content = (String *) &content;
+    godot_string result;
+    memnew_placement(&result, String(_self->insert(at_pos, *_content)));
+    
+    return result;
+}
+
+godot_bool GDAPI godot_string_is_subsequence_of(const godot_string *self, const  godot_string *string) {
+    const String *_self = (const String *) self;
+    const String *_string = (const String *) string;
+
+    return _self->is_subsequence_of(*_string);
+}
+
+godot_bool GDAPI godot_string_is_subsequence_ofi(const godot_string *self, const godot_string *string) {
+    const String *_self = (const String *) self;
+    const String *_string = (const String *) string;
+
+    return _self->is_subsequence_ofi(*_string);
+}
+
+godot_string GDAPI godot_string_lpad(const godot_string *self, godot_int min_length) {
+    const godot_string *default_character = (const godot_string  *) new String(" ");
```

Comment:
This actually leaks memory. So either construct a real new `godot_string` using `godot_string_new{_data}` and then `godot_string_destroy`, but don't just create a new String on the heap and forget about it 😄 

In the case of methods with default parameters you should just call those in C++ directly instead of forwarding to the C version, that way the bindings automatically update if the default parameter in C++ changes.